### ISSUE TITLE
Token lifetimes on organization level

### ISSUE THREAD

**Issue by fforootd** [zitadel/zitadel#5219]
As a administrator, I want to define the lifetime of token not only on instance level but also on the organization level to allow each (business) customer to tweak this according to his own security guidelines.

```[tasklist]
### Acceptance Criteria
- [ ] I can define the token lifetimes on an per org level
- [ ] I nothing is defined the instance defaults are applied
```

**Comment by fforootd** [comment]
This was requested in https://github.com/zitadel/zitadel/discussions/2129#discussioncomment-4972902 and makes total sense to me. Also it feels like a "good first issue", or not?

**Comment by osenchenko** [comment]
I would like to add additional acceptance criteria:

- [ ]  I can define the token lifetimes on an per application level

Use cases:
- Different levels of confidentiality. For video conference systems we can set longer token lifetimes. For example, weeks. But for password managers token lifetimes has to be hours. Not days, or weeks.
- Different devices or different ways how users access applications. For tokens requested from the user's device, we can set longer lifetimes. For web ui, accessed from a browser, shorter lifetimes.

**Comment by fforootd** [comment]
> * token lifetimes on an per application level

I see, that makes sense.

@hifabienne should we add this to this issue or create an additional one for the app level settings?

**Comment by hifabienne** [comment]
> > * token lifetimes on an per application level
> 
> I see, that makes sense.
> 
> @hifabienne should we add this to this issue or create an additional one for the app level settings?

We can add it to this issue and if we find out that it is to big for implementation we can still split it up

**Comment by nielsNocore** [comment]
for us, this would be a very nice feature to have. as our web UI will nog have a very long lifetime in tokens, but some third party data service is allowed to have a much longer (refresh) token lifetime

**Comment by Jank1310** [comment]
Organisation level would be great for different b2b customers which have different requirements

**Comment by dhassoun2021** [comment]
Hello,
Could you tell me when this feature is planned to be implemented in Zitadel?
Thanks and regards

**Comment by hifabienne** [comment]
At the moment we do not have an eta for this feature.

**Comment by osvaldo-azcurra-osanasalud** [comment]
I have a question that may be related to this topic. Is there any attribute in the token that allows us to differentiate whether it was created with a particular application?

**Comment by fforootd** [comment]
> I have a question that may be related to this topic. Is there any attribute in the token that allows us to differentiate whether it was created with a particular application?

Yeah we should have the `azp` claim in the tokens which should indicate which client_id requested it in the first place.

**Comment by osvaldo-azcurra-osanasalud** [comment]
> > I have a question that may be related to this topic. Is there any attribute in the token that allows us to differentiate whether it was created with a particular application?
> 
> Yeah we should have the `azp` claim in the tokens which should indicate which client_id requested it in the first place.

Great! I am using version `v2.51.3` and the `jwt` does not have the `azp` property. In my use case I have an organization with one project and two applications and I cannot distinguish with which application the `jwt` was generated.

**Comment by fforootd** [comment]
> > > I have a question that may be related to this topic. Is there any attribute in the token that allows us to differentiate whether it was created with a particular application?
> > 
> > 
> > Yeah we should have the `azp` claim in the tokens which should indicate which client_id requested it in the first place.
> 
> Great! I am using version `v2.51.3` and the `jwt` does not have the `azp` property. In my use case I have an organization with one project and two applications and I cannot distinguish with which application the `jwt` was generated.

Lets create a new issue for that. Please share the token content and how you retrieved it.

Looking at our docs it should be included https://zitadel.com/docs/apis/openidoauth/claims

**Comment by nasirovsh** [comment]
Would be great to have this feature implemented. We have requirement to have different JWT token lifetime per app

**Comment by vs-gtadeu** [comment]
I also wanted to share that we would benefit from having this implemented.

My only additional request is to also be able to configure the access token lifetime for specific service users.

**Comment by BeHustle1** [comment]
I agree, it would be very nice to be able to manage token lifetimes at the organizational level. In the B2B segment, this is one of the key requests

**Comment by GingerJones** [comment]
I would really like to see this!

**Comment by Drowze** [comment]
Hey Zitadel crew - can we have this supported soon? 🙏 

Just illustrating our scenario, we have under the same project:
* an oidc app for our react frontend app
* an oidc app for our backend (server-to-server)
* an oidc app for our internal swagger ui page
* an oidc app for our internal cli tool

I've looked up solutions to tune their lifetime, but unfortunately it doesn't seem possible to distinguish token lifetime between applications/organizations 😕
